lib/modules/browser_console.js

```diff
@@ -26,12 +26,31 @@ class Log {
     }
 }
 
+function stringifyArg(arg) {
+    return arg.executionContext().evaluate(element => {
+        if (typeof element === 'object' && !(element instanceof RegExp)) {
+            element = JSON.stringify(element);
+        }
+        return String(element);
+    }, arg);
+}
+
+async function stringifyLog(log) {
+    if (log.text().includes('JSHandle@object')) {
+        const args = await Promise.all(log.args().map(stringifyArg));
+        log._text = args.join(' ');
```

Comment:
please, avoid using "private" methods or properties (prepended with  "_") outside of its class, consider only returning the string message in this method and mounting the Log object in the console callback